app/models/game.rb

```diff
@@ -29,29 +29,210 @@ def populate_board!
   end
 
   def in_check?(color)
-    # Query for remaining friendly pieces
-    player_pieces = pieces.where(game_id: id, color: color, captured?: false)
-
-    # Query for enemy king piece
-    enemy_king = pieces.where(game_id: id, type: 'King').where.not(color: color).last
-
-    # Store enemy_king coordinates for shorter reference
-    ekx = enemy_king.x_coordinate
-    eky = enemy_king.y_coordinate
-
-    player_pieces.each do |player_piece|
-      # Check if it would be valid for the current player_piece to move to the enemy king's position
-      if player_piece.valid_move? ekx, eky
-        # Check if the current player_piece wouldn't obstructed while attempting to move to the enemy king's position
-        unless player_piece.is_obstructed? ekx, eky
-          # If both criteria are met, at least once player_piece is successfully 'checking' the enemy king
+    set_remaining_piece_data color
+    
+    @remaining_player_pieces.each do |player_piece|
+    # Check if it would be valid for the current player_piece to move to the opponent king's position
+      if player_piece.valid_move? @okx, @oky
+      # Check if the current player_piece wouldn't obstructed while attempting to move to the opponent king's position
+        unless player_piece.is_obstructed? @okx, @oky
+        # Store reference to player piece causing check
+          @player_piece_causing_check = player_piece
+        # Store player_piece coordinates for shorter reference elsewhere
+          @x_of_threat = player_piece.x_coordinate
+          @y_of_threat = player_piece.y_coordinate
+        # If both criteria are met, at least one player_piece is 'checking' the opponent king
           return true
         end
       end
     end
-    # Return false if no player_piece is 'checking' the enemy king
+
+  # Return false: no player_piece is 'checking' the opponent king
+    false
+  end
+  
+  def ends_by_checkmate?(color)
+  # Return false immediately unless the game is in check
+    return false unless in_check? color
+  # Return false if opponent can capture the piece(s) causing check
+    return false if opponent_can_capture_piece_causing_check? color
+  # Return false if opponent can block the piece(s) causing check
+    return false if opponent_can_block_piece_causing_check? color
+  # Return false if opponent king can escape check
+    return false if checked_king_can_escape?
+
+    true
+  end
+  
+  def opponent_can_capture_piece_causing_check?(color)
```

Comment:
I'm aware 'color' isn't the method name that's why I wrote 'method argument' which you do have as an argument for this method 'opponent_can_capture_piece_causing_check?'. I can see in the method 'ends_by_checkmate?' (line 54) you do use the argument 'color' several times but in this method I don't see it called at all which leaves me to believe that this method doesn't need the 'color' argument. I'm referring to lines 67-87 for this note.